dataactcore/utils/jobQueue.py

```diff
@@ -1,46 +1,80 @@
+    InterfaceHolder directly causes cyclic dependency woes, so we're passing
+    in a class"""
+    with job_app.app_context():
+        job_manager = interface_holder_class().jobDb
+
+        try:
+            rows_of_dicts = fileF.generateFRows(job_manager.session,
+                                                submission_id)
+            header = [key for key in fileF.mappings]    # keep order
+            body = []
+            for row in rows_of_dicts:
+                body.append([row[key] for key in header])
+
+            write_csv(timestamped_name, is_local, header, body)
+            job_manager.markJobStatus(job_id, "finished")
+        except Exception as e:
+            # Log the error
+            job_manager.getJobById(job_id).error_message = str(e)
+            job_manager.markJobStatus(job_id, "failed")
+            job_manager.session.commit()
+
+        job_manager.close()
+
 
 class JobQueue:
     def __init__(self, job_queue_url="localhost"):
         # Set up backend persistent URL
-        backendUrl = ''.join(['db+', CONFIG_DB['scheme'], '://', CONFIG_DB['username'], ':', CONFIG_DB['password'], '@',
-                              CONFIG_DB['host'], '/', CONFIG_DB['job_queue_db_name']])
-
-        # Set up url to the validator for the RESTFul calls
-        validatorUrl = ''.join([CONFIG_SERVICES['validator_host'], ':', str(CONFIG_SERVICES['validator_port'])])
-        if 'http://' not in validatorUrl:
-            validatorUrl = ''.join(['http://', validatorUrl])
+        backendUrl = ('db+{scheme}://{username}:{password}@{host}'
+                      '/{job_queue_db_name}').format(**CONFIG_DB)
 
         # Set up url to the job queue to establish connection
-        queueUrl = ''.join(
-            [CONFIG_JOB_QUEUE['broker_scheme'], '://', CONFIG_JOB_QUEUE['username'], ':', CONFIG_JOB_QUEUE['password'],
-             '@', job_queue_url, ':', str(CONFIG_JOB_QUEUE['port']), '//'])
+        queueUrl = ('{broker_scheme}://{username}:{password}@{host}:{port}'
+                    '//').format(host=job_queue_url, **CONFIG_JOB_QUEUE)
 
         # Create remote connection to the job queue
         self.jobQueue = Celery('tasks', backend=backendUrl, broker=queueUrl)
         self.jobQueue.config_from_object('celeryconfig')
 
-        @self.jobQueue.task(name='jobQueue.enqueue')
-        def enqueue(jobID):
-            # Don't need to worry about the response currently
-            CloudLogger.log("Adding job {} to the queue".format(str(jobID)))
-            url = ''.join([validatorUrl, '/validate/'])
-            params = {
-                'job_id': jobID
-            }
-            response = requests.post(url, params)
-            CloudLogger.log("Job {} has completed validation".format(str(jobID)))
-            CloudLogger.log("Validator response: {}".format(str(response.json())))
-            return response.json()
-
-        self.enqueue = enqueue
```

Comment:
This may require a change to the last line of dataactcore.models.jobTrackerInterface.checkJobDependencies that currently queues validation and cross-file validation jobs that are ready to be started, could you check if that line is consistent with the new way of queueing?  I think the way it's written now is dependent on enqueue being part of JobQueue.